src/api/java/io/github/opencubicchunks/cubicchunks/api/worldgen/populator/WorldGenEntitySpawner.java

```diff
@@ -57,14 +60,21 @@ public static void initialWorldGenSpawn(WorldServer world, Biome biome, int bloc
             Biome.SpawnListEntry currEntry = WeightedRandom.getRandomItem(world.rand, spawnList);
             int groupCount = MathHelper.getInt(random, currEntry.minGroupCount, currEntry.maxGroupCount);
             IEntityLivingData data = null;
-            int randX = blockX + random.nextInt(sizeX);
-            int randZ = blockZ + random.nextInt(sizeZ);
-
-            final int initRandX = randX;
-            final int initRandZ = randZ;
-
+            int randX = blockX + random.nextInt(sizeX) - 1;
+            int randZ = blockZ + random.nextInt(sizeZ) - 1;
+            float yaw = random.nextFloat() * 360.0f - 45.0f;
+            
             for (int i = 0; i < groupCount; ++i) {
                 for (int j = 0; j < 4; ++j) {
+                    randX += random.nextInt(5) + 1;
+                    randZ += random.nextInt(5) + 1;
+                    yaw += random.nextFloat() * 90.0f + 45.0f;
+                    if (randX >= blockX + sizeX)
+                        randX -= sizeX;
+                    if (randZ >= blockZ + sizeZ)
+                        randZ -= sizeZ;
```

Comment:
Why does that work and the vanilla way doesnt?

Reply:
An only option I can think out - random seed which was set based on coordinate hash has some weird properties.